### PR TITLE
Fix a codegen fragment spread bug

### PR DESCRIPTION
**Pull request by mgilson**
The `RELEASE.md` should summarize the bug fix pretty well with an example.  Ultimately, we were assuming that a `GraphQLFragmentSpread` node was not able to appear in a particular position and that assumption was false.

(The previous code had a `# TODO` to figure out how to deal with this -- so seemingly we were aware that this was a potential problem, we just hadn't had time/need to fix it.)

## Description

The main change here is that we effectively handle the `GraphQLFragmentSpread` nodes like we handle them elsewhere.

## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

* #3085

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by botberry** [comment]
Thanks for adding the `RELEASE.md` file!



Here's a preview of the changelog:

---

This fixes a bug where codegen would choke on FragmentSpread nodes in the graphQL during type collection.

e.g.:

```
fragment PartialBlogPost on BlogPost {
  title
}

query OperationName {
  interface {
    id
    ... on BlogPost {
      ...PartialBlogPost
    }
    ... on Image {
      url
    }
  }
}
```

The current version of the code generator is not able to handle the `...PartialBogPost` in this position because it assumes it can only find `Field` type nodes even though the spread should be legit.


---
Here's the preview release card for twitter:




Here's the tweet text: 
```
🆕 Release (next) is out! Thanks to Matt Gilson for the PR 👏

Get it here 👉 https://github.com/strawberry-graphql/strawberry/releases/tag/(next)
```

**Review by patrick91 — APPROVED** [review]

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+Release type: patch
+
+This fixes a bug where codegen would choke on FragmentSpread nodes in the graphQL during type collection.
```
Comment (patrick91) [inline RELEASE.md:3]:
```suggestion
This fixes a bug where codegen would choke on FragmentSpread nodes in the GraphQL during type collection.
```

**Comment by patrick91** [comment]
happy to merge this when pre-commit is green 😊

**Comment by mgilson** [comment]
That precommit makes for an interesting interplay between `black` and `ruff` ... I just changed the text in the error message to avoid messing with it 😅

**Pull request merged**